ca_tools/scripts/pos_controller_pid.py

```diff
@@ -0,0 +1,190 @@
+#!/usr/bin/env python
+
+import rospy
+import math
+from nav_msgs.msg import Odometry
+from geometry_msgs.msg import Twist
+from tf.transformations import euler_from_quaternion
+from geometry_msgs.msg import Pose2D
+
+
+class DrawSquare:
+
+    # Time variables
+    Now=0.0
```

Comment:
I recommend you to declare here variables that are not going to be modified throughout the code. And in `__init__` variables that need to be initialized.